src/lib/setup/subscriptionService.ts

```diff
@@ -0,0 +1,45 @@
+import { SubscriptionClient } from "@azure/arm-subscriptions";
+import {
+  ApplicationTokenCredentials,
+  loginWithServicePrincipalSecret
+} from "@azure/ms-rest-nodeauth";
+import { logger } from "../../logger";
+import { IRequestContext } from "./constants";
+
+export interface ISubscriptionItem {
+  id: string;
+  name: string;
+}
+
+/**
+ * Returns a list of subscriptions based on the service principal credentials.
+ *
+ * @param rc Request Context
+ */
+export const getSubscriptions = (
+  rc: IRequestContext
+): Promise<ISubscriptionItem[]> => {
```

Comment:
why not using `async` since the entire code base using it? `Promise` is used in few places due to SDK dependency.

Reply:
can you show me how you want the code to be written? async keyword is needed when we have await in the function; and wrapping the entire function as a Promise

Reply:
it is already returning `promise` which means you are using asynchronous or blocking code in the function. `async` functions use an implicit Promise to return results anyway. Keeping code consistent would be good.

Reply:
my point is we do not add async unnecessary and it is not encourage. it is not about consistency.
Sorry I do not get your point on why you want to add async?

Reply:
I can write the function in the manner but not simply adding an async keyword.
```
export const getSubscriptions = async (
  rc: IRequestContext
): Promise<ISubscriptionItem[]> => {
  logger.info("attempting to get subscription list");
  const creds = await loginWithServicePrincipalSecret(
    rc.servicePrincipalId!,
    rc.servicePrincipalPassword!,
    rc.servicePrincipalTenantId!
  );
  const client = new SubscriptionClient(creds);
  const subsciptions = await client.subscriptions.list();
  const result = (subsciptions || []).map(s => {
    return {
      id: s.subscriptionId!,
      name: s.displayName!
    };
  });
  logger.info("Successfully acquired subscription list");
  return result;
};
```